devito/symbolics/extended_sympy.py

```diff
@@ -156,6 +163,49 @@ def base(self):
     __reduce_ex__ = Pickable.__reduce_ex__
 
 
+class FieldFromPointer(FunctionFromPointer, Pickable):
+
+    """
+    Symbolic representation of the C notation ``pointer->field``.
+    """
+
+    def __new__(cls, field, pointer):
+        return FunctionFromPointer.__new__(cls, field, pointer)
+
+    def __str__(self):
+        return '%s->%s' % (self.pointer, self.field)
+
+    @property
+    def field(self):
+        return self.function
+
+    __repr__ = __str__
+
+
+class FieldFromComposite(FunctionFromPointer, Pickable):
+
+    """
+    Symbolic representation of the C notation ``composite.field``,
+    where ``composite`` is a struct/union/...
+    """
+
+    def __new__(cls, field, composite):
+        return FunctionFromPointer.__new__(cls, field, composite)
+
+    def __str__(self):
+        return '%s.%s' % (self.composite, self.field)
+
+    @property
+    def field(self):
+        return self.function
+
+    @property
+    def composite(self):
+        return self.pointer
+
+    __repr__ = __str__
+
+
```

Comment:
Does these two really belong here?

Reply:
wdym?

Reply:
FunctionFromPointer isn't a sympy utility, this would be more in function or utils ather than sympy extras no?

Reply:
it is, it inherits from SymPy types